File: resto-pulse-mvp/frontend/lib/online-order-categories.ts
export const ONLINE_ORDER_CATEGORIES = [
  {
    slug: 'tatli-tuzlu',
    label: 'Tatlı & Tuzlu',
    hint: 'Pasta, pastane, baklava, tatlıcı',
  },
  { slug: 'doner', label: 'Döner', hint: 'Et veya tavuk döner' },
  {
    slug: 'kebap-izgara',
    label: 'Kebap & Izgara',
    hint: 'Adana, Urfa, köfte, steak, ızgara kanat/tavuk',
  },
  { slug: 'firin', label: 'Fırın', hint: 'Pizza, lahmacun, pide' },
  {
    slug: 'sokak',
    label: 'Sokak Lezzetleri',
    hint: 'Çiğ köfte, tantuni, tost, sokak yemeği',
  },
  { slug: 'burger', label: 'Burger', hint: 'Burger ve fast food' },
  {
    slug: 'ev-yemekleri',
    label: 'Ev Yemekleri',
    hint: 'Ev yemekleri, mantı, makarna, dünya mutfağı',
  },
  { slug: 'kahvalti', label: 'Kahvaltı', hint: 'Kahvaltı mekanları' },
  { slug: 'kahve', label: 'Kahve & İçecek', hint: 'Kahve, çay, içecek mekanları' },
  { slug: 'deniz', label: 'Deniz Ürünleri', hint: 'Balık ve deniz mahsulleri' },
  {
    slug: 'salata-fit',
    label: 'Salata & Fit',
    hint: 'Salata, meze, vejetaryen, fit menü',
  },
] as const;

export const LEGACY_CATEGORY_SLUG_MAP: Record<string, string> = {
  'pasta-tatli': 'tatli-tuzlu',
  'pastane-firin': 'tatli-tuzlu',
  baklava: 'tatli-tuzlu',
  'kahve-icecek': 'kahve',
  tavuk: 'kebap-izgara',
  kebap: 'kebap-izgara',
  kofte: 'kebap-izgara',
  steak: 'kebap-izgara',
  pizza: 'firin',
  'pide-lahmacun': 'firin',
  'cig-kofte': 'sokak',
  tantuni: 'sokak',
  'tost-sandwich': 'sokak',
  'sokak-lezzetleri': 'sokak',
  'manti-makarna': 'ev-yemekleri',
  'dunya-mutfagi': 'ev-yemekleri',
  'deniz-urunleri': 'deniz',
  salata: 'salata-fit',
  meze: 'salata-fit',
  vejetaryen: 'salata-fit',
  fit: 'salata-fit',
};

export function normalizeCategorySlug(slug: string): string {
  const key = slug.trim().toLowerCase();
  return LEGACY_CATEGORY_SLUG_MAP[key] ?? key;
}

File: resto-pulse-mvp/backend/app/constants/online_order_categories.py
"""Online siparis — mutfak etiketleri (daraltılmış liste)."""

from __future__ import annotations

ONLINE_ORDER_CATEGORIES: tuple[dict[str, str], ...] = (
    {"slug": "tatli-tuzlu", "label": "Tatlı & Tuzlu"},
    {"slug": "doner", "label": "Döner"},
    {"slug": "kebap-izgara", "label": "Kebap & Izgara"},
    {"slug": "firin", "label": "Fırın"},
    {"slug": "sokak", "label": "Sokak Lezzetleri"},
    {"slug": "burger", "label": "Burger"},
    {"slug": "ev-yemekleri", "label": "Ev Yemekleri"},
    {"slug": "kahvalti", "label": "Kahvaltı"},
    {"slug": "kahve", "label": "Kahve & İçecek"},
    {"slug": "deniz", "label": "Deniz Ürünleri"},
    {"slug": "salata-fit", "label": "Salata & Fit"},
)

VALID_CATEGORY_SLUGS = frozenset(row["slug"] for row in ONLINE_ORDER_CATEGORIES)

LEGACY_CATEGORY_SLUG_MAP: dict[str, str] = {
    "pasta-tatli": "tatli-tuzlu",
    "pastane-firin": "tatli-tuzlu",
    "baklava": "tatli-tuzlu",
    "kahve-icecek": "kahve",
    "tavuk": "kebap-izgara",
    "kebap": "kebap-izgara",
    "kofte": "kebap-izgara",
    "steak": "kebap-izgara",
    "pizza": "firin",
    "pide-lahmacun": "firin",
    "cig-kofte": "sokak",
    "tantuni": "sokak",
    "tost-sandwich": "sokak",
    "sokak-lezzetleri": "sokak",
    "manti-makarna": "ev-yemekleri",
    "dunya-mutfagi": "ev-yemekleri",
    "deniz-urunleri": "deniz",
    "salata": "salata-fit",
    "meze": "salata-fit",
    "vejetaryen": "salata-fit",
    "fit": "salata-fit",
}


def normalize_category_slugs(raw: list[str] | None) -> list[str]:
    if not raw:
        return []
    seen: set[str] = set()
    out: list[str] = []
    for item in raw:
        slug = (item or "").strip().lower()
        slug = LEGACY_CATEGORY_SLUG_MAP.get(slug, slug)
        if not slug or slug not in VALID_CATEGORY_SLUGS or slug in seen:
            continue
        seen.add(slug)
        out.append(slug)
    return out


def category_label(slug: str) -> str:
    for row in ONLINE_ORDER_CATEGORIES:
        if row["slug"] == slug:
            return row["label"]
    legacy = LEGACY_CATEGORY_SLUG_MAP.get(slug)
    if legacy:
        return category_label(legacy)
    return slug
